### PR TITLE
feat(api): support custom api host

### PR DESCRIPTION
**Pull request by lianghx-319**
In some countries, it is too hard to accessing OpenAI. So they can deploy it's own reverse proxy server to access openai API

### CODE REVIEW

**Review by jackMort — CHANGES_REQUESTED** [review]
What do you think about adding it to the configuration rather than as an environment variable? You can still use your logic when setting up the plugin on your end

**Comment by lianghx-319** [comment]
> 

I think it is same as `OPEN_API_KEY`, the host maybe treat as some kind of secret. So if allow to setup in configuration, maybe implement like `OPEN_API_KEY` is good. Set at environment is the easy way to make thing works. I will spend more time to support it to the configuration if necessary.

File: lua/chatgpt/api.lua [hunk 1]
```diff
@@ -4,10 +4,12 @@ local logger = require("chatgpt.common.logger")
 
 local Api = {}
 
+-- API Host please no http or https
+Api.OPENAI_API_HOST = os.getenv("OPENAI_API_KEY") or "api.openai.com"
```
Comment (2nthony) [inline lua/chatgpt/api.lua:8]: is this should be `os.getenv("OPENAI_API_HOST")` ?

Comment (lianghx-319) [reply lua/chatgpt/api.lua:8]: Yes, thanks.

File: lua/chatgpt/api.lua [hunk 2]
```diff
@@ -151,39 +146,58 @@ local splitCommandIntoTable = function(command)
...
-  if not Api.OPENAI_API_KEY then
-    if Config.options.api_key_cmd ~= nil and Config.options.api_key_cmd ~= "" then
-      loadApiKey(Config.options.api_key_cmd)
+local function loadConfigFromEnv(envName, configName)
+  Api[configName] = os.getenv(envName)
+  if not Api[configName] then
+    logger.warn(envName .. " environment variable not set")
+    return
+  end
+  Api[configName] = Api[configName]:gsub("%s+$", "")
```
Comment (lianghx-319) [inline lua/chatgpt/api.lua:172]: `  local variable = os.getenv(envName)
  if not variable then
    logger.warn(envName .. " environment variable not set")
    return
  end
  Api[configName] = variable:gsub("%s+$", "")`

Comment (lianghx-319) [reply lua/chatgpt/api.lua:172]: should not log warning in this method, because configuration should get from plugin option as a fallback.

**Comment by lianghx-319** [comment]
> What do you think about adding it to the configuration rather than as an environment variable? You can still use your logic when setting up the plugin on your end

api host has aded to the configuration, can you review the final code implement?

**Comment by you-n-g** [comment]
Very exciting to see such PR.
When would this PR be merged?

Thanks!

**Comment by lianghx-319** [comment]
@jackMort Are you have time to have a look?

**Review by jackMort — APPROVED** [review]

**Pull request merged**